Add earliest housing demographics provider

spamhurts opened the pull request:
Added additional housing query to link to demographics customizer

spamhurts committed:
Added earliest housing demographics provider and query.

spamhurts committed:
removed redundant column wrapping method and replaced "Id" strings with a constant

spamhurts committed:
actually removed the redundant column wrapping method that was missed on the last commit and aligned wrapped column name with the demographic provider

RameshRapa commented on snprc_ehr/resources/queries/study/demographicsEarliestHousing.sql:
```diff
@@ -0,0 +1,17 @@
+
+
+--date of first housing
+LEFT JOIN study.housing as h on h.id = d.id and h.qcstate.publicdata = true
```
Hi Terry, Can i please know why do we need to check for this condition in the query? "qcstate.publicdata = true".

spamhurts replied:
Public data are rows that have passed QC checks. In the EHR, data that has not been verified should not be available for viewing.

jallentxbiomed commented on snprc_ehr/src/org/labkey/snprc_ehr/table/SNPRC_EHRCustomizer.java:
```diff
@@ -62,6 +64,8 @@ public class SNPRC_EHRCustomizer extends AbstractTableCustomizer
 
     private Set<CalculatedColumn> calculatedColumns;
 
+    public static final String ID_COL = "Id";
```
In src/org/labkey/snrpc_ehr/constants/QueryConstants.java we have this string already as constant ID_COLUMN

spamhurts replied:
Thanks. I'll make the change.

jallentxbiomed commented on snprc_ehr/src/org/labkey/snprc_ehr/table/SNPRC_EHRCustomizer.java:
```diff
@@ -161,8 +165,14 @@ public void doTableSpecificCustomizations(AbstractTableInfo ti)
         {
+            UserSchema us = getEHRUserSchema(ti, STUDY_SCHEMA);
             customizeAnimalTable(ti);
+            var col42 = getWrappedCol(us, ti, "earliestHousing", "demographicsEarliestHousing",ID_COL,ID_COL);
```
I'm not sure that it's clear what col42 means as a variable name here

spamhurts replied:
I'm just following the naming scheme used in the defaultEHRCustomizer

spamhurts replied:
It's just being used locally - I'll change it to col

spamhurts committed:
- Use the ID_COLUMN constant in QueryConstants.java.
 - Renamed local wrapped column var name.
- added offspring to the browse all datasets screen

RameshRapa commented on snprc_ehr/src/org/labkey/snprc_ehr/table/SNPRC_EHRCustomizer.java:
```diff
@@ -167,10 +163,10 @@ public void doTableSpecificCustomizations(AbstractTableInfo ti)
-            col42.setLabel("Housing - Earliest");
-            col42.setDescription("The calculates the earliest housing location for each living animal.");
-            ti.addColumn(col42);
+            var col = getWrappedCol(us, ti, "earliestHousing", "demographicsEarliestHousing", ID_COLUMN, ID_COLUMN);
```
Hi Terry. the import for ID_COLUMN is missing.

spamhurts replied:
Look again - it's on line 48

spamhurts replied:
Jay added that import previously

spamhurts committed:
commented out earliest housing registration to sort out test issue
